drivers/modem/hl7800.c

```diff
@@ -4999,6 +5003,72 @@ static int setup_gprs_connection(char *access_point_name)
 	return send_at_cmd(NULL, cmd_string, MDM_CMD_SEND_TIMEOUT, 0, false);
 }
 
+static int set_bands(const char *bands, bool full_reboot)
+{
+	int ret;
+	char cmd[sizeof("AT+KBNDCFG=#,####################")];
```

Comment:
`bands` should have a max length of `MDM_BAND_BITMAP_STR_LENGTH_MAX` in `mdm_hl7800_set_bands` which is 2 + 6 + 14 = 22, but here only 20 chars are allocated?

Reply:
No, 4+8+8 = 20.